add-dbs/embed.cpp

```diff
@@ -0,0 +1,245 @@
+                              static_cast<void*>(&configs),
+                              nullptr);
+            if(rc != SQLITE_OK)
+            {
+                std::cerr << "Unable to execute query: " << query_ss.str() << std::endl;
+                exit(-1);
+            }
+            if(arch_idx != 0)
+                ss << " else ";
+
+            ss << "if(arch_cu == \"" << arch << "_" << num_cu << "\") \n {\n static const "
+                                                                 "std::unordered_map<std::string, "
+                                                                 "std::string> data \n\t {\n";
+            // handle the joining comma
+            ss << "\t\t { \"" << configs[0]["key"] << "\", \"" << configs[0]["res"] << "\"}"
+               << std::endl;
+            for(auto cfg_idx = 1; cfg_idx < configs.size(); cfg_idx++)
+            {
+                auto config = configs[cfg_idx];
+                if(!config["res"].empty())
+                    ss << "\t\t,{ \"" << config["key"] << "\", \"" << config["res"] << "\"}"
+                       << std::endl;
+            }
+            ss << "};\n return data; \n}" << std::endl;
+        }
+        // default case for invalid/unknown arch-cu pair
+        ss << "else {\n static const std::unordered_map<std::string, std::string> data \n\t {\n}; "
+              "\n return data; \n}"
+           << std::endl;
+    }
+    else // plaintext
+    {
+
+        auto line = std::string{};
+        std::ofstream ss(output_file.string(), std::ios::out);
+        if(!ss.good())
+        {
+            std::cerr << "Unable to open output file: " << output_file.string() << std::endl;
+            exit(-1);
+        }
+        for(auto arch_idx = 0; arch_idx < arches.size(); arch_idx++)
+        {
+            auto kinder = arches[arch_idx];
+            auto arch   = kinder.first;
+            auto num_cu = kinder.second;
+
+            boost::filesystem::path input_file;
+            for(auto& tmp : input_files)
+            {
+                std::string str_file = boost::filesystem::basename(tmp);
+                std::stringstream tmp_ss;
+                tmp_ss << arch << "_" << num_cu;
+                if(str_file.find(tmp_ss.str()) != std::string::npos)
+                {
+                    input_file = tmp;
+                    break;
+                }
+            }
+            auto file = std::ifstream{input_file.string()};
+            if(!file)
+            {
+                std::cerr << "Unable to open input file: " << input_file.string() << std::endl;
+                exit(-1);
+            }
+            if(arch_idx != 0)
+                ss << " else ";
+
+            ss << "if(arch_cu == \"" << arch << "_" << num_cu << "\") \n {\n static const "
+                                                                 "std::unordered_map<std::string, "
+                                                                 "std::string> data \n\t {\n";
```

Comment:
I am not sure, @pfultz2 probably knows better here, but I think this should not be static, and you should move the result out. It is only ever called once on the first ReadonlyRamDb request, and static is just a plus one hanging an in-memory copy of the data.